hash.c

```diff
@@ -0,0 +1,40 @@
+#include <linux/crc32.h>
+#include <linux/hash.h>
+#include <linux/stringhash.h>
+#include <linux/types.h>
+#include "config.h"
+#include "simplefs.h"
+
+
+#if defined(CONFIG_SIMPLEFS_HASH_HASH64)
+static uint64_t _hash64(const char *name, size_t len)
```

Comment:
Is it really necessary for us to re-implement our own hash function? Upstream Linux already has many hash implementations, like full_name_hash(). Could we use those here instead?